### PR TITLE
Adds the whisper_function file

### PR DESCRIPTION
**Pull request by AdamMutzell**
Här kan du importera funktionen transcribe som gör det vi vill

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+numpy
+openai-whisper
+torch==2.3.0
```
Comment (Uplink036) [inline requirements.txt:3]: Either we should add a specific version to all of them or none of them. I vote for all of them, even though they should work for now and into the future.

File: whisper_functions.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        return "Error"
+
+    return result["text"]
+
+if __name__ == "__main__":
+    device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    # Load the model
+    model = whisper.load_model("turbo").to(device)
+
+    audio_dir = "data/archive/Raw JL corpus (unchecked and unannotated)/JL(wav+txt)"
```
Comment (Uplink036) [inline whisper_functions.py:64]: A bit hard coded, but not sure what more we can do now. Think it's a good idea to break this out now into a function or shall we leave it till we are more done with the RL model?

File: whisper_functions.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    result_df = pd.DataFrame(columns=["file", "annotation", "transcription"])
+    for file in os.listdir(audio_dir):
+        if file.endswith(".wav"):
```
Comment (Uplink036) [inline whisper_functions.py]: Did we make a commitment to .waw files? I don't remember if we did, though all of this data is .waw. Probably should make an issue / documentation about it if that's the case.

File: whisper_functions.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            file_path = os.path.join(audio_dir, file)
+            transcription = transcribe(model, file_path, cuda=device == 'cuda')
+
+            annotation_file = file.replace(".wav", ".txt")
```
Comment (Uplink036) [inline whisper_functions.py]: Interesting solution... If we did make this into a function, I would say we don't limit ourselves to files and filenames like this.

Comment (Uplink036) [reply whisper_functions.py]: If we want to leave it like this for now, which i am fine with, I would like a comment in the code or as issue for the future.

File: whisper_functions.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            if os.path.exists(os.path.join(audio_dir, annotation_file)):
+                with open(os.path.join(audio_dir, annotation_file), "r") as f:
+                    annotation = f.read()
+            else:
+                annotation = None
+
+            result_df.loc[len(result_df)] = [file, annotation, transcription]
+    
+    result_df.to_csv("data/transcriptions.csv", index=False)
```
Comment (Uplink036) [inline whisper_functions.py]: This should also be a function in the future with a good output.

Comment (Uplink036) [reply whisper_functions.py]: Might even want it to be one now.

**Review by Uplink036 — CHANGES_REQUESTED** [review]
I would like the comments to be talked about or fixed before we move merge. Preferably through documentation like github.

File: Makefile [hunk 6]
```diff
@@ -1,7 +1,10 @@
 dependencies: ## Install all project requirements
+	sudo apt update && sudo apt install ffmpeg -Y
```
Comment (Uplink036) [inline Makefile]: Small '-y' not '-Y'.

File: requirements.txt (hunk 1 above)
Comment (AdamMutzell) [reply requirements.txt:3]: Specifik till alla blir bra

File: whisper_functions.py (hunk 2 above)
Comment (AdamMutzell) [reply whisper_functions.py:64]: Allt är hårdkodat för att pröva det som ett exempel. Filen är framförallt för funktioner kopplade till Whisper

File: whisper_functions.py (hunk 3 above)
Comment (AdamMutzell) [reply whisper_functions.py]: Ja vi pratade om det. .wav blir nog enklast i.o.m. att det är det vi har fått. Vi kan dokumentera det

File: whisper_functions.py (hunk 4 above)
Comment (AdamMutzell) [reply whisper_functions.py]: Vet inte hur det annars ska göras. Igen, bara ett test, inte generellt.

File: whisper_functions.py (hunk 5 above)
Comment (AdamMutzell) [reply whisper_functions.py]: Ja självklart

File: Makefile (hunk 6 above)
Comment (AdamMutzell) [reply Makefile]: Fixed

**Review by AdamMutzell — COMMENTED** [review]
I've pushed up some changes

File: whisper_functions.py (hunk 2 above)
Comment (Uplink036) [reply whisper_functions.py:64]: I like the new changes.

**Review by Uplink036 — APPROVED** [review]
It's better now, still some things like documentation and versions, but it will come with time.

**Pull request merged**